content/docs/user-guide/hydra-integration.md

```diff
@@ -0,0 +1,251 @@
+(`conf/config.yaml`):
+
+```dvc
+$ cat conf/config.yaml
+defaults:
+  - dataset: imagenette
+  - train: baseline
+```
+
+<admon type="info">
+
+You can set the [`dvc config hydra`](/doc/command-reference/config#hydra)
+options to provide custom locations for the _Config Group_ and the top-level
+_Defaults List_.
+
+</admon>
+
+In this example, we also have a nested Defaults List:
+
+```dvc
+$ cat conf/train/baseline.yaml
+defaults:
+  - model: resnet
+  - optimizer: sgd
+```
+
+## Setting up DVC
+
+We can enable the DVC and Hydra integration by running:
+
+```dvc
+$ dvc config hydra.enabled True
+```
+
+### Testing the integration
+
+To check how the integration works, we can create a dummy `dvc.yaml` that prints
+the contents of `params.yaml`:
+
+```yaml
+stages:
+  print-params:
+    cmd: cat params.yaml
+```
+
+Now, if we run an experiment, we will see how the composed configuration looks:
+
+```dvc
+$ dvc exp run
+Running stage 'print-params':
+> cat params.yaml
+dataset:
+  url: https://s3.amazonaws.com/fast-ai-imageclas/imagenette2-160.tgz
+  output_folder: imagenette
+train:
+  model:
+    name: ResNet
+    size: 50
+    weights: ResNet50_Weights.IMAGENET1K_V2
+  optimizer:
+    name: SGD
+    lr: 0.001
+    momentum: 0.9
+```
+
+### Building a Pipeline
+
+To showcase a more realistic use case, we are going to build a Pipeline with 2
+stages.
+
+The first stage downloads the dataset, and depends on the
+<abbr>parameters</abbr> defined in the `dataset` section:
+
+```yaml
+stages:
+  setup-dataset:
+    cmd:
+    - wget ${dataset.url} -O tmp.tgz
+    - mkdir -p ${dataset.output_folder}
+    - tar zxvf tmp.tgz -C ${dataset.output_folder}
+    - rm tmp.tgz
+    outs:
+      - ${dataset.output_folder}
+  ...
+```
+
+We use [Templating](/doc/user-guide/project-structure/dvcyaml-files#templating)
+in order to configure our shell commands (`mkdir`, `tar` and `wget`) and to
+avoid hardcoding the <abbr>output</abbr> paths.
+
+The second stage trains a model using the downloaded dataset and depends on the
+rest of parameters:
+
+```yaml
+  ...
+  train:
+    cmd: python train.py ${train}
+    deps:
+      - ${dataset.output_folder}
+```
+
+We use
+[Dict Unpacking](/doc/user-guide/project-structure/dvcyaml-files#dict-unpacking)
+in order to pass our configuration as
+[argparse](https://docs.python.org/3/library/argparse.html) arguments.
+
+<admon type="tip">
+
+[Dict Unpacking](/doc/user-guide/project-structure/dvcyaml-files#dict-unpacking)
+can be also used with other languages. For example, we could use
+[R argparse](https://cran.r-project.org/web/packages/argparse/vignettes/argparse.html)
+, [Julia ArgParse](https://argparsejl.readthedocs.io/en/latest/argparse.html) or
+any other shell command that accepts argparse-like syntax.
+
+</admon>
+
+As an alternative, we could (only for Python scripts) use the
+`dvc.api.params_show()` method in order to load all the required parameters
+inside the stage:
```

Comment:
Is there a way to move this up? This seems the most straightforward way, and it's closest to a drop-in replacement for @hydra.main.

Also, can you add in the corresponding `dvc.yaml` section to show how to track the `params` there?

Reply:
I added a `tab` section with the Python API being the default tab shown